issue_test.go

```diff
@@ -538,6 +538,26 @@ func TestIssueService_PostAttachment_NoAttachment(t *testing.T) {
 	}
 }
 
+func TestIssueService_DeleteAttachment(t *testing.T) {
+	setup()
+	defer teardown()
+	testMux.HandleFunc("/rest/api/2/attachment/10054", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "DELETE")
+		testRequestURL(t, r, "/rest/api/2/attachment/10054")
+
+		w.WriteHeader(http.StatusNoContent)
+		fmt.Fprint(w, `{}`)
+	})
+
+	resp, err := testClient.Issue.DeleteAttachment("10054")
+	if resp.StatusCode != 204 {
+		t.Error("Expected attachment not deleted.")
+	}
+	if err != nil {
+		t.Errorf("Error given: %s", err)
+	}
+}
+
```

Comment:
Can you please add negative tests? to ensure that the `if` branches for error checking are validated?

Reply:
So when err == nil or StatusCode == 404, 403?

Please add edits so I can better understand.